src/Temporalio.Extensions.Hosting/TemporalWorkerServiceId.cs

```diff
@@ -0,0 +1,12 @@
+namespace Temporalio.Extensions.Hosting
+{
+    /// <summary>
+    /// Represents the unique identifier for a hosted Temporal Worker service.
+    /// </summary>
+    public record TemporalWorkerServiceId(
+        string TaskQueue,
+        string? Version,
+        bool IsBuildId)
```

Comment:
```suggestion
        bool VersionIsBuildId)
```

Made it a bit clearer to me, but not a big deal. Alternatively could accept optional and mutually exclusive build ID _and_ `WorkerDeploymentVersion`, e.g.:

```csharp
public record TemporalWorkerServiceIdentifier(
    string TaskQueue,
    string? UnversionedBuildId,
    WorkerDeploymentVersion? Version)
{
    public TemporalWorkerServiceIdentifier(string taskQueue, string? unversionedBuildId = null)
        : this(taskQueue, unversionedBuildId, null)
    {
    }

    public TemporalWorkerServiceIdentifier(string taskQueue, WorkerDeploymentVersion version)
        : this(taskQueue, null, version)
    {
    }
}
```

But not a big deal

Reply:
Eh, they're not really mutually exclusive though. It's no safer.

Reply:
Just a bit more confusing that `Version` isn't `Version`, it's sometimes `BuildId`. Maybe then just the boolean property change is enough.